sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/BufferSendState.scala

```diff
@@ -171,31 +181,36 @@ class BufferSendState(
 
         acquiredBuffs.foreach { case RangeBuffer(blockRange, rapidsBuffer) =>
```

Comment:
Currently the caller (when UCX callback is triggered) calls `releaseAcquiredToCatalog`, and we require this state is cleaned in https://github.com/NVIDIA/spark-rapids/pull/1876/files#diff-039d2429369bd2d3796876c1f91257f62836765e46509a6e7abc9534df22abecR146. It means the behavior of this is to allow the catalog to spill the buffer while we are working a window and waiting for possibly long amount of times. You are changing that behavior here not just for GDS, but for all tiers.

If you have a bunch of transfers in flight and each is holding onto a large shuffle block that a bounce buffer window is sliding through, I think we are going to run into trouble if we hold each of those large shuffle blocks for the entire duration of that transfer.

Were you able to check if copying from GDS to device memory at an offset is possible? And is it performant for bounce-buffer length copies? To me that seems ideal, the buffer isn't materialized wholesale, and it arrives at memory that is destined to be used for the transfer.